lib/proxy.ts

```diff
@@ -608,7 +581,7 @@ export class Proxy implements IProxy {
                       port,
                     };
                     hosts.forEach((host) => {
-                      self.sslServers[hostname] = sslServer;
+                      self.sslServers[host] = sslServer;
```

Comment:
this was the bug that I mentioned in the previous diff. I *think* this is correct, but would appreciate a second look